doc/README.md

```diff
@@ -0,0 +1,421 @@
+and to provide backwards compatibility.
+
+Blobs include the raw bytes of the compressed chunks
+and snapshots their compressed protobuf encoding.
+
+```console
+┏━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┓
+┃ Version ┃ encrypted tink payload ┃
+┃   0x02  ┃ (with 40 bytes header) ┃
+┗━━━━━━━━━┻━━━━━━━━━━━━━━━━━━━━━━━━┛
+```
+
+## Snapshots
+
+Encoded [in protobuf format](../app/src/main/proto/snapshot.proto).
+Example printed as JSON:
+
+```json
+{
+  "token": 171993168400,
+  "name": "Google Pixel 7a",
+  "androidId": "bbcc5909347d0b83",
+  "sdkInt": 34,
+  "androidIncremental": "eng.user.20240618.130805",
+  "d2d": true,
+  "apps": {
+    "org.example.app": {
+      "time": 171993268400,
+      "state": "apkAndData",
+      "type": "full",
+      "name": "Example app",
+      "system": false,
+      "launchableSystemApp": false,
+      "chunkIds": [
+        "001b527ebb5eb57f4934bafeb998cb08595ed7ced603d9d25bd3c50b338f939d",
+        "ff2e9f9c75d211602c5a68f0471aa549e2499a3fa9496255b26678f4aad75a98"
+      ],
+      "apk": {
+        "versionCode": 2342,
+        "installer": "org.fdroid.basic",
+        "signatures": [
+          "abd81091c552574506bbb143e3bd856504382666dd495811a539188957522ab2"
+        ],
+        "splits": [
+          {
+            "name": "base",
+            "size": 1337,
+            "chunkIds": [
+              "4f34352be7c1f4d1597d27f5824b42bad2222cf029cc888e9d7318d5e8bc2ad5",
+              "16759b52e8ef8c3b080f0a8cdb8ac21474cb41070f272a02e28fd1163762336b"
+            ]
+          }
+        ]
+      }
+    }
+  },
+  "iconChunkIds": [
+    "c11c3cf1c326375b177e768de908e6e423c3d8866715c8ec7159156b6ea82497"
+  ],
+  "blobs": {
+    "001b527ebb5eb57f4934bafeb998cb08595ed7ced603d9d25bd3c50b338f939d": {
+      "id": "259bb78b72b7b7f7ac1a9613c3c4936b06342471b39b1c41a328acaa2d3ccca5",
+      "length": 645312,
+      "uncompressedLength": 695312
+    },
+    "ff2e9f9c75d211602c5a68f0471aa549e2499a3fa9496255b26678f4aad75a98": {
+      "id": "5e6a8789571183a97e84f74fe13c4e95b6c5645e2d53fe5cc7aa122897c246b4",
+      "length": 9455314,
+      "uncompressedLength": 9855314
+    },
+    "4f34352be7c1f4d1597d27f5824b42bad2222cf029cc888e9d7318d5e8bc2ad5": {
+      "id": "a79d2b9e7afa051782cff6424ede1337c8ce05c34d9c2e565e2ed47e7ae80ea4",
+      "length": 8430346,
+      "uncompressedLength": 9430346
+    },
+    "16759b52e8ef8c3b080f0a8cdb8ac21474cb41070f272a02e28fd1163762336b": {
+      "id": "31aa8e88c8b73df565fd8bbcadfe7e2e79303eb5c0223253353d3003273e3140",
+      "length": 26549052,
+      "uncompressedLength": 28549052
+    },
+    "c11c3cf1c326375b177e768de908e6e423c3d8866715c8ec7159156b6ea82497": {
+      "id": "5421895dbead0ba83f0993f17b5f72f6c6618c0e9ea849c6c53ac240d1802d0b",
+      "length": 523496,
+      "uncompressedLength": 645314
+    }
+  }
+}
+```
+
+The `chunkIds` and `iconChunkIds` fields contain a list with plain text SHA-256 hashes
```

Comment:
[Clarity] It's probably self-explanatory, but consider adding a short description of what iconChunkIds refers to.